### PR TITLE
Update to prefect v1.2.0

### LINKED ISSUE #28: Kindly update codebase with new version of Prefect

### PR DESCRIPTION
**Pull request by mikebirdgeneau**
- Updates files to Prefect version 1.2.0.
- Server `docker-compose.yml` updated based on the output from `prefect server config` when run with version 1.2.0.
- Minor update to imports in weather.py example.

Resolves #28

### CODE REVIEW

File: server/docker-compose.yml [hunk 1]
```diff
@@ -1,95 +1,143 @@
...
+      hasura:
+        condition: service_started
     environment:
-      PREFECT_SERVER_DB_CMD: ${PREFECT_SERVER_DB_CMD:-"echo 'DATABASE MIGRATIONS SKIPPED'"}
-      PREFECT_SERVER__DATABASE__CONNECTION_URL: ${DB_CONNECTION_URL}
-      PREFECT_SERVER__HASURA__ADMIN_SECRET: ${PREFECT_SERVER__HASURA__ADMIN_SECRET:-hasura-secret-admin-secret}
+      PREFECT_CORE_VERSION: 1.2.0
+      PREFECT_SERVER_DB_CMD: prefect-server database upgrade -y
+      PREFECT_SERVER__DATABASE__CONNECTION_URL: postgresql://${POSTGRES_USER}:${POSTGRES_PASSWORD}@postgres:5432/${POSTGRES_DB}
+      PREFECT_SERVER__HASURA__ADMIN_SECRET: ${HASURA_GRAPHQL_ADMIN_SECRET}
```
Comment (flavienbwk) [inline server/docker-compose.yml:66]: This value is missing in `server/.env`, we could replace it by a default value with :

```suggestion
      PREFECT_SERVER__HASURA__ADMIN_SECRET: ${PREFECT_SERVER__HASURA__ADMIN_SECRET:-hasura-secret-admin-secret}
```

**Review by flavienbwk — CHANGES_REQUESTED** [review]

File: server/docker-compose.yml [hunk 2]
```diff
@@ -1,95 +1,143 @@
...
   towel:
-    image: "prefecthq/server:core-0.15.3"
-    command: "python src/prefect_server/services/towel/__main__.py"
+    command: python src/prefect_server/services/towel/__main__.py
+    depends_on:
+      graphql:
+        condition: service_started
     environment:
-      PREFECT_SERVER__HASURA__ADMIN_SECRET: ${PREFECT_SERVER__HASURA__ADMIN_SECRET:-hasura-secret-admin-secret}
+      PREFECT_SERVER__HASURA__ADMIN_SECRET: ${HASURA_GRAPHQL_ADMIN_SECRET}
```
Comment (flavienbwk) [inline server/docker-compose.yml:86]:
```suggestion
      PREFECT_SERVER__HASURA__ADMIN_SECRET: ${PREFECT_SERVER__HASURA__ADMIN_SECRET:-hasura-secret-admin-secret} 
```

**Review by flavienbwk — CHANGES_REQUESTED** [review]
Sorry forgot this one.

**Comment by flavienbwk** [comment]
Thank you very much for your contribution @mikebirdgeneau !

**Comment by mikebirdgeneau** [comment]
Thanks for sharing this clean example!
I was about to make my own, and found this. :+1: 

Cheers!

**Pull request merged**